### DISCUSSION TITLE
Temperature syntax

### DISCUSSION THREAD

**Discussion by dnlzro** [cooklang/spec#67] [Ideas]
There doesn't seem to be any special syntax for specifying a temperature in the current specification. Having special syntax would make it easier to highlight this information in a GUI. It would also make temperature conversion possible for people who prefer to use a certain measurement system.

**Comment by dubadub** [comment]
@1izardo I was thinking about the temperature but didn't find a way to use this information in GUI hence didn't add anything. 

And you offered a very valid use case for temperature conversion 👍, it makes sense to have it. Do you have any suggestions on syntax?

**Comment by dnlzro** [comment]
@dubadub I was thinking something like `@{350%F}` or `@{200%C}`, for example, since it follows similar syntax as timers and it is naturally read as "at this temperature," but that might conflict with how ingredients are defined. Maybe use a `^` instead of `@` for temperature?

**Comment by tobei** [comment]
> @dubadub I was thinking something like `@{350%F}` or `@{200%C}`, for example, since it follows similar syntax as timers and it is naturally read as "at this temperature," but that might conflict with how ingredients are defined. Maybe use a `^` instead of `@` for temperature?

It is frequent to need both temperature and duration so maybe it would help to have both as separate syntax or allow both in the brackets as long as it's possible to start a timer from it.

**Comment by wpearse** [comment]
An interesting point to note: 350 F is a common baking temperature, but the direct conversion to ~176 C does not make sense: if you are cooking and your oven has a thermostat that's set in deg C you will use 180 C.

For this reason, might I suggest that temperatures can be specified in both F and C at the same time? 

`@{350%F,180%C}`

This is laborious, although the alternative might be a conversion table that maps 350 F to 180 C.

**Reply by nicholaswilde** [reply]
> This is laborious, although the alternative might be a conversion table that maps 350 F to 180 C.

Conversion table can be added to the config file perhaps.

**Reply by dubadub** [reply]
I think it's a good idea to add it to conversion table

**Reply by solpyro** [reply]
Would it make sense to round to 2SF?

**Comment by briansunter** [comment]
I would like to have a temperature syntax as well. It's relevant to me for baking and temperature units conversion. Something like `^{350%C}` `^{350%F}` `!{350%C}` etc

**Comment by tobei** [comment]
Alternatively can just let whatever cooklang client do the conversion into the user's prefered unit and rouding settings. Clients can let the user choose between automatically displaying original unit, exact converted units, author supplied conversion or using any internal table.

**Reply by jnmclarty** [reply]
This is the way.  Let the computers to the math.

**Comment by jnmclarty** [comment]
Would it be possible to pick up temperatures using `*C` or `*F`, prefixed with a number between -999 and 999?

So `380*F`  would be all you need to write?

**Reply by jnmclarty** [reply]
I would imagine the token you could hunt for would be `*F` or `*C` , and even if there was a space before it, you could pick up the number right before quite elegantly.  Eg. `380 *F` or even a sloppy `380  *F` with two spaces would work.

**Reply by solpyro** [reply]
`(\d+)\s?[\*°][CF]` should do the trick. Although it'll fall down when people use the gas mark system

**Reply by jphastings** [reply]
I'd appreciate a syntax as close to human writing as possible (less to remember how to write), so this seems best (accepting `*` _and_ `º`)

**Comment by ckafi** [comment]
It seems this discussion has fizzled out. Is there any new input on this? I would really appreciate it if I would no longer have to convert temperatures all the time.

**Comment by philocalyst** [comment]
I'm a fan of the * syntax!

**Reply by Earnestly** [reply]
Yes, if `°` is also accepted as @jphastings noted.

**Comment by michalmoc** [comment]
I'd suggest to extend it to other units as well. It's not that rare to have recipe ask to cut something in slices of some centimeters, or use for now only 50dag out of full 1kg of ingredients.

**Comment by tomodachi94** [comment]
Something that should probably be defined in the temperature spec is the that the Unicode degrees symbol (`U+00B0`/`°`/`degree sign`) should be used instead of graphically similar symbols like `U+00BA`/`º`/`masculine ordinal indicator`.

Additionally, `U+2103`/`℃`/`degree Celsius` and `U+2109`/`℉`/`degree Fahrenheit` should be discouraged, per the Unicode Standard:

> In normal use, it is better to represent degrees Celsius “°C” with a sequence of U+00B0 degree sign + U+0043 latin capital letter c, rather than U+2103 degree celsius. For searching, treat these two sequences as identical. Similarly, the sequence U+00B0 degree sign + U+0046 latin capital letter f is preferred over U+2109 degree fahrenheit,
and those two sequences should be treated as identical for searching.

**All of this does also assume that Unicode is the preferred character encoding for `.cook` files, which isn't explicitly defined anywhere that I could find.**

**Comment by gvlx** [comment]
No need for another syntax as time and temperature bring their own markers so something like `#oven{1%h}{20%C}` or `#mixer{280%F}{10%m}` is easily parsed.
